### PR TITLE
refactor gisaid to fit new files

### PR DESCRIPTION
**Pull request by Mropat**
## Description
*This PR adds/fixes ...*

### How to prepare for test
- [ ] ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] paxa the environment: `paxa`
- [ ] install on stage (example for hasta):
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-cg-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [ ] Document in ...
- [ ] Deploy this branch
- [ ] Inform to ...

### CODE REVIEW

**Comment by mayabrandi** [comment]
Testing on hasta:

```
[0|0|0] [maya.brandi@hasta:/home/proj/stage/mutant] [S_main] 1m6s 1 $ cg upload gisaid maturejay
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
----------------- UPLOAD ----------------------
----------------- GISAID UPLOAD -------------------
Fetch latest version from bundle maturejay
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/bin/conda/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Fetching files with tags in [komplettering]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [komplettering]
Fetching files from version 44204
Creating GisaidSample for ACC7910A3
Creating GisaidSample for ACC7910A5
Creating GisaidSample for ACC7910A72
Creating GisaidSample for ACC7910A6
Creating GisaidSample for ACC7910A81
Creating GisaidSample for ACC7910A68
Creating GisaidSample for ACC7910A52
Creating GisaidSample for ACC7910A64
Creating GisaidSample for ACC7910A76
Creating GisaidSample for ACC7910A75
Creating GisaidSample for ACC7910A69
Creating GisaidSample for ACC7910A58
Creating GisaidSample for ACC7910A8
Creating GisaidSample for ACC7910A15
Creating GisaidSample for ACC7910A2
Creating GisaidSample for ACC7910A92
Creating GisaidSample for ACC7910A80
Creating GisaidSample for ACC7910A83
Creating GisaidSample for ACC7910A74
Creating GisaidSample for ACC7910A82
Creating GisaidSample for ACC7910A7
Creating GisaidSample for ACC7910A66
Creating GisaidSample for ACC7910A88
Creating GisaidSample for ACC7910A18
Creating GisaidSample for ACC7910A9
Creating GisaidSample for ACC7910A94
Creating GisaidSample for ACC7910A63
Creating GisaidSample for ACC7910A93
Creating GisaidSample for ACC7910A10
Creating GisaidSample for ACC7910A59
Creating GisaidSample for ACC7910A71
Creating GisaidSample for ACC7910A49
Creating GisaidSample for ACC7910A13
Creating GisaidSample for ACC7910A57
Creating GisaidSample for ACC7910A67
Creating GisaidSample for ACC7910A84
Creating GisaidSample for ACC7910A12
Creating GisaidSample for ACC7910A22
Creating GisaidSample for ACC7910A77
Creating GisaidSample for ACC7910A55
Creating GisaidSample for ACC7910A11
Creating GisaidSample for ACC7910A61
Creating GisaidSample for ACC7910A85
Creating GisaidSample for ACC7910A50
Creating GisaidSample for ACC7910A86
Creating GisaidSample for ACC7910A78
Creating GisaidSample for ACC7910A91
Creating GisaidSample for ACC7910A73
Creating GisaidSample for ACC7910A89
Creating GisaidSample for ACC7910A70
Creating GisaidSample for ACC7910A14
Creating GisaidSample for ACC7910A25
Creating GisaidSample for ACC7910A16
Creating GisaidSample for ACC7910A56
Creating GisaidSample for ACC7910A51
Creating GisaidSample for ACC7910A47
Creating GisaidSample for ACC7910A90
Creating GisaidSample for ACC7910A79
Creating GisaidSample for ACC7910A19
Creating GisaidSample for ACC7910A60
Creating GisaidSample for ACC7910A20
Creating GisaidSample for ACC7910A62
Creating GisaidSample for ACC7910A24
Creating GisaidSample for ACC7910A26
Creating GisaidSample for ACC7910A23
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-csv,maturejay]
Fetching files from version 44204
GISAID CSV for case maturejay exists, will be replaced
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-fasta,maturejay]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A3,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A5,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A72,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A6,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A81,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A68,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A52,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A64,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A76,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A75,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A69,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A58,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A8,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A15,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A2,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A92,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A80,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A83,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A74,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A82,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A7,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A66,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A88,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A18,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A9,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A94,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A63,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A93,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A10,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A59,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A71,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A49,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A13,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A57,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A67,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A84,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A12,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A22,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A77,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A55,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A11,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A61,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A85,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A50,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A86,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A78,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A91,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A73,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A89,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A70,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A14,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A25,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A16,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A56,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A51,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A47,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A90,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A79,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A19,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A60,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A20,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A62,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A24,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A26,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [ACC7910A23,consensus-sample]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-log,maturejay]
Fetching files from version 44204
GISAID log exists in latest bundle in Housekeeper
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-csv,maturejay]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-fasta,maturejay]
Fetching files from version 44204
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-log,maturejay]
Fetching files from version 44204
Running command /home/proj/stage/servers/resources/gisaid_uploader.py --logfile /home/proj/stage/mutant/gisaid/tmpc8o1tm95 CoV upload --csv /home/proj/stage/housekeeper-bundles/maturejay/2021-06-30/maturejay.csv --fasta /home/proj/stage/housekeeper-bundles/maturejay/2021-06-30/maturejay.fasta
Fetch latest version from bundle maturejay
Fetching files with tags in [komplettering]
Fetching files from version 44204
Parsing accesion numbers from log file
Fetch latest version from bundle maturejay
Fetching files with tags in [gisaid-log,maturejay]
Fetching files from version 44204
{'21CS502113': ' EPI_ISL_2812291', '21CS502115': ' EPI_ISL_2812292', '21CS502182': ' EPI_ISL_2812293', '21CS502116': ' EPI_ISL_2812294', '21CS502191': ' EPI_ISL_2812295', '21CS502178': ' EPI_ISL_2812296', '21CS502162': ' EPI_ISL_2812297', '21CS502174': ' EPI_ISL_2812298', '21CS502186': ' EPI_ISL_2812299', '21CS502185': ' EPI_ISL_2812300', '21CS502179': ' EPI_ISL_2812301', '21CS502168': ' EPI_ISL_2812302', '21CS502118': ' EPI_ISL_2812303', '21CS502125': ' EPI_ISL_2812304', '21CS502112': ' EPI_ISL_2812305', '21CS501864': ' EPI_ISL_2812306', '21CS502190': ' EPI_ISL_2812307', '21CS502193': ' EPI_ISL_2812308', '21CS502184': ' EPI_ISL_2812309', '21CS502192': ' EPI_ISL_2812310', '21CS502117': ' EPI_ISL_2812311', '21CS502176': ' EPI_ISL_2812312', '21CS502198': ' EPI_ISL_2812313', '21CS502128': ' EPI_ISL_2812314', '21CS502119': ' EPI_ISL_2812315', '21CS501866': ' EPI_ISL_2812316', '21CS502173': ' EPI_ISL_2812317', '21CS501865': ' EPI_ISL_2812318', '21CS502120': ' EPI_ISL_2812319', '21CS502169': ' EPI_ISL_2812320', '21CS502181': ' EPI_ISL_2812321', '21CS502159': ' EPI_ISL_2812322', '21CS502123': ' EPI_ISL_2812323', '21CS502167': ' EPI_ISL_2812324', '21CS502177': ' EPI_ISL_2812325', '21CS502194': ' EPI_ISL_2812326', '21CS502122': ' EPI_ISL_2812327', '21CS502132': ' EPI_ISL_2812328', '21CS502187': ' EPI_ISL_2812329', '21CS502165': ' EPI_ISL_2812330', '21CS502121': ' EPI_ISL_2812331', '21CS502171': ' EPI_ISL_2812332', '21CS502195': ' EPI_ISL_2812333', '21CS502160': ' EPI_ISL_2812334', '21CS502196': ' EPI_ISL_2812335', '21CS502188': ' EPI_ISL_2812336', '21CS501863': ' EPI_ISL_2812337', '21CS502183': ' EPI_ISL_2812338', '21CS501861': ' EPI_ISL_2812339', '21CS502180': ' EPI_ISL_2812340', '21CS502124': ' EPI_ISL_2812341', '21CS502135': ' EPI_ISL_2812342', '21CS502126': ' EPI_ISL_2812343', '21CS502166': ' EPI_ISL_2812344', '21CS502161': ' EPI_ISL_2812345', '21CS502157': ' EPI_ISL_2812346', '21CS501862': ' EPI_ISL_2812347', '21CS502189': ' EPI_ISL_2812348', '21CS502129': ' EPI_ISL_2812349', '21CS502170': ' EPI_ISL_2812350', '21CS502130': ' EPI_ISL_2812351', '21CS502172': ' EPI_ISL_2812352', '21CS502134': ' EPI_ISL_2812353', '21CS502136': ' EPI_ISL_2812354', '21CS502133': ' EPI_ISL_2812355'}
Upload to GISAID successful

```

File: cg/meta/upload/gisaid/gisaid.py [hunk 1]
```diff
@@ -74,131 +107,152 @@ def get_gisaid_samples(self, case_id: str) -> List[GisaidSample]:
...
-        sample_rows = []
         for sample in gisaid_samples:
-            sample_row: List[str] = self.get_sample_row(gisaid_sample=sample)
-            sample_rows.append(sample_row)
-        return sample_rows
+            fasta_file: File = self.housekeeper_api.find_file_in_latest_version(
+                case_id=case_id, tags=[sample.cg_lims_id, "consensus-sample"]
+            )
+            if not fasta_file:
+                LOG.warning(f"No fasta file found for sample {sample.cg_lims_id}")
```
Comment (mayabrandi) [inline cg/meta/upload/gisaid/gisaid.py:128]: I forgot... Why would we continue if a fasta file is missing? Is this handeled when building the csv file? Or will there be a missmach between fasta and csv then? Will gisaid acceppt that?

File: cg/meta/upload/gisaid/gisaid.py [hunk 2]
```diff
@@ -14,14 +14,9 @@
 from cg.store import models, Store
 from cg.utils import Process
 from .constants import HEADERS
-from .models import GisaidSample, UploadFiles, GisaidAccession
-import csv
+from .models import GisaidSample, GisaidAccession
```
Comment (mayabrandi) [inline cg/meta/upload/gisaid/gisaid.py:17]: UploadFiles can be deleted. No?

File: cg/meta/upload/gisaid/gisaid.py [hunk 3]
```diff
@@ -207,103 +261,31 @@ def get_accession_numbers(self, log_file: Path) -> Dict[str, str]:
...
-                sample_id = sample["provnummer"]
-                selection_criteria = sample["urvalskriterium"]
-                completion_nr = completion_data.get(sample_id, "")
-                new_completion_file_data.append([sample_id, selection_criteria, completion_nr])
-        return new_completion_file_data
-
-    def update_completion_file(
-        self, completion_file: Path, new_completion_file_data: List[List[str]]
-    ) -> None:
+    def update_completion_file(self, case_id: str) -> None:
```
Comment (mayabrandi) [inline cg/meta/upload/gisaid/gisaid.py:266]: :) 👍  a lot cleaner with pandas!

**Review by mayabrandi — APPROVED** [review]
Great! 😊

File: cg/meta/upload/gisaid/gisaid.py (hunk 1 above)
Comment (Mropat) [reply cg/meta/upload/gisaid/gisaid.py:128]: good point

File: cg/meta/upload/gisaid/gisaid.py [hunk 4]
```diff
@@ -74,131 +109,167 @@ def get_gisaid_samples(self, case_id: str) -> List[GisaidSample]:
...
+        """Create all gisaid files in Housekeeper, if needed."""
+        gisaid_samples = self.get_gisaid_samples(case_id=case_id)
+        self.create_gisaid_csv(gisaid_samples=gisaid_samples, case_id=case_id)
+        self.create_gisaid_fasta(gisaid_samples=gisaid_samples, case_id=case_id)
+        self.create_gisaid_log_file(case_id=case_id)
 
-        with open(str(gisaid_log.absolute()), "w") as open_gisaid_log:
-            json.dump(new_log_data, open_gisaid_log)
-        temp_log.unlink()
+    def authenticate_gisaid(self):
```
Comment (mayabrandi) [inline cg/meta/upload/gisaid/gisaid.py:199]: nice

**Comment by Mropat** [comment]
```
Successfully built cg
Installing collected packages: cg
  Attempting uninstall: cg
    Found existing installation: cg 22.8.0
    Uninstalling cg-22.8.0:
      Successfully uninstalled cg-22.8.0
Successfully installed cg-22.8.1
```

**Pull request merged**